### PR TITLE
fix: homogenize authoring format and place

### PR DESCRIPTION
**Pull request by davorpa**
## What does this PR do?
Add info | Improve repo

### Description

- use colon as unique author delimiter (replaces `&`, `and`, `y`, `e`, `et`...)
- ensures is placed after resource title, not part of itself `title - author`. Strip `by`
- `et al.` as special author/token to group many
- ~remove author role title/acronyms~
- use ~`trad.`~ `trl.` as traducer notation instead remove it (needs review all added resources) Definition list: https://loc.gov/marc/relators/relaterm.html
- remove delimiters when denotes an empty text author: `,,` `- (` `, (`...

## Checklist:
- [x] Read our [contributing guidelines](https://github.com/EbookFoundation/free-programming-books/blob/main/docs/CONTRIBUTING.md).
- [x] Include author(s) and platform where appropriate.
- [x] Put lists in alphabetical order, correct spacing.
- [x] Add needed indications (PDF, access notes, under construction).
- [x] Used an informative name for this pull request.

## Follow-up

- Check the status of GitHub Actions and resolve any reported warnings!

### CODE REVIEW

**Comment by davorpa** [comment]
31dc7bb5ef1905d11e3b46fe71232f66a2dcd53d resolves conflicts with #7032 merge

File: books/free-programming-books-fr.md [hunk 1]
```diff
@@ -203,24 +203,24 @@
...
 #### PGF/TikZ
 
-* [TikZ pour l'impatient](http://math.et.info.free.fr/TikZ/) - Gérard Tisseau et Jacques Duma
+* [TikZ pour l'impatient](http://math.et.info.free.fr/TikZ/) - Gérard Tisseau, Jacques Duma
 
 
 #### TeX
 
 * [Apprendre à programmer en TeX](https://ctan.org/pkg/apprendre-a-programmer-en-tex) - Christian Tellechea
-* [TeX pour l'Impatient](http://www.apprendre-en-ligne.net/LaTeX/teximpatient.pdf) - Paul Abrahams, Kathryn Hargreaves, and Karl Berry, trad. Marc Chaudemanche (PDF)
```
Comment (eshellman) [inline books/free-programming-books-fr.md:212]: the translator notation should stay

File: books/free-programming-books-fr.md [hunk 2]
```diff
@@ -309,23 +309,23 @@
...
 
 ### Systèmes d'exploitation
 
-* [Simple OS (SOS)](http://sos.enix.org/fr/SOSDownload) - David Decotigny et Thomas Petazzoni
+* [Simple OS (SOS)](http://sos.enix.org/fr/SOSDownload) - David Decotigny, Thomas Petazzoni
 
 
 ### TEI
 
-* [Qu'est-ce que la Text Encoding Initiative ?](http://books.openedition.org/oep/1237) - Lou Burnard, trad. par Marjorie Burghart
```
Comment (eshellman) [inline books/free-programming-books-fr.md:328]: keep a translator notation

File: books/free-programming-books-hu.md [hunk 3]
```diff
@@ -24,7 +24,7 @@
 * [Az adatbázistervezés alapjai és titkai](http://mek.oszk.hu/11100/11123/index.phtml) - Halassy Béla (Word, PDF)
 * [Ember, információ, rendszer](http://mek.oszk.hu/11100/11122/index.phtml) - Halassy Béla (Word, PDF)
 * [Formális nyelvek](http://mek.oszk.hu/05000/05099/index.phtml) - Bach Iván (PDF)
-* [Mese a felhasználó központú tervezőről](http://mek.oszk.hu/11700/11748/index.phtml) - David Travis, ford.: Favorit Fordító Iroda (PDF)
```
Comment (eshellman) [inline books/free-programming-books-hu.md:27]: is this a translator notation

File: books/free-programming-books-hu.md [hunk 4]
```diff
@@ -35,7 +35,7 @@
 
 ### Arduino
 
-* [Arduino programozási kézikönyv](http://avr.tavir.hu) - Brian W. Evans írása alapján fordította, kiegészítette és frissítette Cseh Róbert (PDF - regisztráció szükséges)
```
Comment (eshellman) [inline books/free-programming-books-hu.md:38]: not sure what this means

File: books/free-programming-books-hu.md [hunk 5]
```diff
@@ -51,7 +51,7 @@
 ### HTML and CSS
 
 * [CSS alapjai](http://weblabor.hu/cikkek/cssalapjai1) - Bártházi András (HTML)
-* [Webes szabványok](http://nagygusztav.hu/webes-szabvanyok) - Chris Mills, Ben Buchanan, Tom Hughes-Croucher, Mark Norman "Norm" Francis, Linda Goin, Paul Haine, Jen Hanen, Benjamin Hawkes-Lewis, Ben Henick, Christian Heilmann, Roger Johansson, Peter-Paul Koch, Jonathan Lane, Tommy Olsson, Nicole Sullivan és Mike West, ford.: Nagy Gusztáv (PDF)
```
Comment (eshellman) [inline books/free-programming-books-hu.md:54]: maybe it's an editor notation? should keep

File: books/free-programming-books-hu.md [hunk 6]
```diff
@@ -107,10 +107,10 @@
 
 ### Python
 
-* [Bevezetés a Pythonba példákkal](http://mek.oszk.hu/08400/08436/index.phtml) - Raphaël Marvie, ford.: Daróczy Péter (PDF)
```
Comment (eshellman) [inline books/free-programming-books-hu.md:110]: keep role notations

File: books/free-programming-books-it.md [hunk 7]
```diff
@@ -242,7 +242,7 @@
 
 * [Il manuale di riferimento di Python](http://docs.python.it/html/ref/)
 * [Il tutorial di Python](http://docs.python.it/html/tut/)
-* [Immersione in Python 3](http://gpiancastelli.altervista.org/dip3-it/) - Mark Pilgrim, _traduzione di: Giulio Piancastelli_ (HTML) [(PDF)](http://gpiancastelli.altervista.org/dip3-it/d/diveintopython3-it-pdf-latest.zip)
```
Comment (eshellman) [inline books/free-programming-books-it.md:245]: role notation

File: books/free-programming-books-pt_BR.md [hunk 8]
```diff
@@ -99,9 +99,9 @@
 
 * [Apostila Linguagem C](http://www.ime.usp.br/~slago/slago-C.pdf) - Silvio Lago (PDF)
 * [Curso de Programação em C - UFMG](http://www2.dcc.ufmg.br/disciplinas/pc/source/introducao_c_renatocm_deeufmg.pdf) - Renato Cardoso Mesquita (PDF)
-* [Guia Beej's Para Programação em Rede - Usando Internet Sockets](http://beej.us/guide/bgnet/translations/bgnet_ptbr.html) - Brian "Beej Jorgensen" Hall, Traduzido por cv8minix3 (HTML)
```
Comment (eshellman) [inline books/free-programming-books-pt_BR.md:102]: role notation

File: casts/free-podcasts-screencasts-en.md [hunk 9]
```diff
@@ -168,7 +168,7 @@
 * [Frontend First](https://player.fm/series/frontend-first) - Sam Selikoff, Ryan Toronto (podcast)
 * [Frontend Five](https://frontendfive.codeschool.com) - CodeSchool (podcast)
 * [HTML All The Things](https://www.htmlallthethings.com/podcast) - Mike Karan, Matt Lawrence (podcast)
-* [JavaScript Air](https://javascriptair.com) - Kent C. Dodds, Dan Abramov, Matt Zabriskie, Pam Selle, Lin Clark, Brian Lonsdorf, Iheanyi Ekechukwu,, Tyler McGinnis, Kyle Simpson, et al.(podcast)
+* [JavaScript Air](https://javascriptair.com) - Kent C. Dodds, Dan Abramov, Matt Zabriskie, Pam Selle, Lin Clark, Brian Lonsdorf, Iheanyi Ekechukwu,, Tyler McGinnis, Kyle Simpson, et al. (podcast)
```
Comment (eshellman) [inline casts/free-podcasts-screencasts-en.md:171]: remove double comma, too?

File: courses/free-courses-en.md [hunk 10]
```diff
@@ -298,11 +298,11 @@
 * [Data Analysis and Visualization](https://www.udacity.com/course/data-analysis-and-visualization--ud404) - Georgia Tech (Udacity)
 * [Data Analysis with Python: Zero to Pandas](https://jovian.ai/learn/data-analysis-with-python-zero-to-pandas) (Jovian)
 * [Data Analysis with R](https://www.udacity.com/course/data-analysis-with-r--ud651) - Facebook (Udacity)
-* [Data Cleaning by Rachael Tatman at Kaggle](https://www.kaggle.com/learn/data-cleaning)
+* [Data Cleaning](https://www.kaggle.com/learn/data-cleaning) - Rachael Tatman (Kaggle)
```
Comment (eshellman) [inline courses/free-courses-en.md:301]: Kaggle here is an affiliation, I think, which we omit

File: courses/free-courses-en.md [hunk 11]
```diff
@@ -298,11 +298,11 @@
...
 * [Data Mining with Weka MOOC](https://www.cs.waikato.ac.nz/ml/weka/mooc/dataminingwithweka/)
 * [Data science for beginners](https://microsoft.github.io/Data-Science-For-Beginners) - Microsoft
 * [Data Science interview questions](https://www.youtube.com/playlist?list=PLZoTAELRMXVPkl7oRvzyNnyj1HS4wt2K-) - Krish Naik
-* [Data Visualization by Alexis Cook at Kaggle](https://www.kaggle.com/learn/data-visualization)
```
Comment (eshellman) [inline courses/free-courses-en.md:305]: affiliation?

**Review by eshellman — CHANGES_REQUESTED** [review]
big PR!

File: books/free-programming-books-fr.md (hunk 1 above)
Comment (davorpa) [reply books/free-programming-books-fr.md:212]: I understand, but in the most of files 98% the traducer is part of the names without any role label.

There are in `hu` `it` `en` `fr`

is `trad.` enough for all file locales?

File: books/free-programming-books-hu.md (hunk 3 above)
Comment (davorpa) [reply books/free-programming-books-hu.md:27]: yes i think, or the editorial

```
A fordítás alapja:
David Travis – The Fable of the User-Centred Designer
2009 USERFOCUS

Fordította: Favorit Fordító Iroda, 2012
```

File: books/free-programming-books-hu.md (hunk 4 above)
Comment (davorpa) [reply books/free-programming-books-hu.md:38]: ... `based on her writing, it was translated, supplemented and updated by Róber Cseh`

File: books/free-programming-books-hu.md (hunk 5 above)
Comment (davorpa) [reply books/free-programming-books-hu.md:54]: Definitely `ford.` is the hungarian notation to translator

File: books/free-programming-books-it.md (hunk 7 above)
Comment (davorpa) [reply books/free-programming-books-it.md:245]: I'll homogenize it across files with `trad.:`

File: casts/free-podcasts-screencasts-en.md (hunk 9 above)
Comment (davorpa) [reply casts/free-podcasts-screencasts-en.md:171]:
```suggestion
* [JavaScript Air](https://javascriptair.com) - Kent C. Dodds, Dan Abramov, Matt Zabriskie, Pam Selle, Lin Clark, Brian Lonsdorf, Iheanyi Ekechukwu, Tyler McGinnis, Kyle Simpson, et al. (podcast)
```

File: courses/free-courses-en.md (hunk 10 above)
Comment (davorpa) [reply courses/free-courses-en.md:301]: It's the courses platform like Sololearn, Coursera, Edx, Udemy...

File: courses/free-courses-en.md (hunk 11 above)
Comment (davorpa) [reply courses/free-courses-en.md:305]: See https://github.com/EbookFoundation/free-programming-books/pull/7034#discussion_r953995859

File: books/free-programming-books-fr.md (hunk 1 above)
Comment (eshellman) [reply books/free-programming-books-fr.md:212]: one possibility is to use `(trl)` https://www.loc.gov/marc/relators/relaterm.html

File: courses/free-courses-en.md (hunk 10 above)
Comment (eshellman) [reply courses/free-courses-en.md:301]: OK

File: books/free-programming-books-hu.md (hunk 5 above)
Comment (davorpa) [reply books/free-programming-books-hu.md:54]: Recovered at 0a4dbc179e4b2f6346fc15579d7c5edec1b3433e

File: books/free-programming-books-hu.md (hunk 6 above)
Comment (davorpa) [reply books/free-programming-books-hu.md:110]: Recovered at 0a4dbc179e4b2f6346fc15579d7c5edec1b3433e

File: books/free-programming-books-it.md (hunk 7 above)
Comment (davorpa) [reply books/free-programming-books-it.md:245]: Recovered at 0a4dbc179e4b2f6346fc15579d7c5edec1b3433e

File: books/free-programming-books-pt_BR.md (hunk 8 above)
Comment (davorpa) [reply books/free-programming-books-pt_BR.md:102]: Recovered at 0a4dbc179e4b2f6346fc15579d7c5edec1b3433e

File: books/free-programming-books-hu.md (hunk 4 above)
Comment (davorpa) [reply books/free-programming-books-hu.md:38]: Recovered at 0a4dbc179e4b2f6346fc15579d7c5edec1b3433e

File: books/free-programming-books-hu.md (hunk 3 above)
Comment (davorpa) [reply books/free-programming-books-hu.md:27]: Recovered at 0a4dbc179e4b2f6346fc15579d7c5edec1b3433e

File: books/free-programming-books-fr.md (hunk 2 above)
Comment (davorpa) [reply books/free-programming-books-fr.md:328]: Recovered at 0a4dbc179e4b2f6346fc15579d7c5edec1b3433e

File: books/free-programming-books-fr.md (hunk 1 above)
Comment (davorpa) [reply books/free-programming-books-fr.md:212]: Recovered at 0a4dbc179e4b2f6346fc15579d7c5edec1b3433e

Comment (davorpa) [reply books/free-programming-books-fr.md:212]: > one possibility is to use `(trl)` or `trl.` [loc.gov/marc/relators/relaterm.html](https://www.loc.gov/marc/relators/relaterm.html)

Applied in 1833c6db7a3d67788f5bda88506d7071c16cd61c

File: casts/free-podcasts-screencasts-en.md (hunk 9 above)
Comment (davorpa) [reply casts/free-podcasts-screencasts-en.md]: Fixed at 2aeeae0799ded988a4565b3a206472979c3ba497

**Comment by davorpa** [comment]
Conflicts are with #7033 merge which add a new QT resource which uses some modified line as hunk

File: books/free-programming-books-subjects.md [hunk 12]
```diff
@@ -661,7 +661,7 @@ Kerridge (PDF) (email address *requested*, not required)
 
 ### Partial Evaluation
 
-* [Partial Evaluation and Automatic Program Generation](http://www.itu.dk/people/sestoft/pebook/) - (PDF) Jones, Gomard and Sestoft
+* [Partial Evaluation and Automatic Program Generation](http://www.itu.dk/people/sestoft/pebook/) - (PDF) Neil D. Jones, C.K. Gomard, Peter Sestoft
```
Comment (davorpa) [inline books/free-programming-books-subjects.md:664]: Sort author - format

```suggestion
* [Partial Evaluation and Automatic Program Generation](http://www.itu.dk/people/sestoft/pebook/) - Neil D. Jones, C.K. Gomard, Peter Sestoft (PDF)
```

**Comment by davorpa** [comment]
Looks fine after merge with #7026 in 9f493144de8cef467792bf5fcda884be689bdf77

**Pull request merged**